### ISSUE TITLE
Expose a modular health check endpoint

### ISSUE THREAD

**Issue by cpackingham** [linkerd/linkerd2#5106]
### What problem are you trying to solve?
The information contained in `linkerd check` is really handy for knowing the status of an installation, but that command doesn't seem to be meant for monitoring by itself. 
```
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API
```
Look at all that good stuff! 

### How should the problem be solved?
 A health check endpoint containing the same information as linkerd check in a JSON format would help in setting up monitoring for linkerd. This allows for flexibility of monitoring implementations for people consuming it, as just about any monitoring tool should have an api check available. If the tool allows you to inspect the response, using a JSON format would allow drilling down further into the health of components for specific alerts.

### Any alternatives you've considered?

- adding exit code 1 on warning to the `linkerd check` command to use for monitoring 

This is potentially low effort but also not very clean and doesn't solve for being able to look for specific status checks.

- built-in alert manager configs

This wouldn't work for folks who don't have the flexibility to use multiple alerting solutions (such as myself). Having to vet and use another alerting tool could be hard. 

### How would users interact with this feature?
Sending a GET request to a control plane component (where does this live?), such as `https://linkerd-x/status/components` and receiving a response with a structure such as this:
```json
{
  "linkerd-existence": {
    "linkerd-config-cm-exists": true,
    "heatbeat-serviceaccount-exists": true,
    ...
  },
  "linkerd-config": {
    "control-plane-namespace-exists": true,
    ...
  }
  ... 
}
```
Ideally the response would contain all of the same information as `linkerd check`, just encoded into JSON.

**Comment by cpackingham** [comment]
^ Also, I would like to implement this!

**Comment by grampelberg** [comment]
@alpeb you wouldn't happen to know where we got with check output in the dashboard? Is that something that'd work for this or would we need more?

**Comment by cpackingham** [comment]
@grampelberg Can you assign this one to me?

**Comment by alpeb** [comment]
Yes, you can leverage the same API endpoint the dashboard uses: `/api/check` exposed by the `linkerd-web` service, which already returns its results in json (the CLI command `linkerd check` itself can also return results in json using the flag `-ojson`).

Also note that `linkerd check` should already have an exit value other than zero for checks whose failure are marked as fatal, such as some control plane serviceaccount not being present.

**Comment by cpackingham** [comment]
The info from `/api/check` looks perfect! I wonder if we need this issue then? Maybe it would be a good thing to figure out how to utilize then make a documentation section on it?
